### ISSUE TITLE
SafetyAssuredComment does not match the correct comment

### ISSUE THREAD

**Issue by garrettblehm** [jeromedalbert/rubocop-obsession#7]
The following code "violates" the `` check.

```ruby
    # Safe because this method does not actually need a safety_assured block
    safety_assured do
      change_column_null :users, :name, true
    end
```

```bash
C: Obsession/Rails/SafetyAssuredComment: Add # Safe because <reason> comment above safety_assured. An invalid reason may bring the site down.
    safety_assured do ...
```

The problem is the line `previous_comment = processed_source.comments_before_line(node.first_line)&.first` is matching a comment above this code. Instead it needs to be `.last`.

**Comment by jeromedalbert** [comment]
Thanks for the report! I am not able to reproduce the issue on my local with a newly created Rails 8 app with rubocop 1.68.0 though. 🤔 

Would you mind running `bundle env` and sharing the full output? So I can see your rubocop version, ruby version, rails version, etc.

Or better yet, you could make a [minimal example app](https://www.codetriage.com/example_app) that repros the issue.

**Comment by jeromedalbert** [comment]
Thanks for the report! I am not able to repro the issue with your example code as is, but I am able to repro if I add an extra `# Hello` comment to the migration:

```ruby
# Hello
class TestRubocopObsession < ActiveRecord::Migration[8.0]
  def change
    # Safe because blah blah blah blah
    safety_assured do
      change_column_null :users, :name, true
    end
  end
end
```

**Comment by jeromedalbert** [comment]
Released version 0.1.10 with the fix.

Let me know if you find any other issues. Thanks!
